packages/dds/tree/src/feature-libraries/chunked-forest/uniformChunk.ts

```diff
@@ -0,0 +1,431 @@
+    public constructor(private readonly chunk: UniformChunk) {
+        super();
+        this.shape = this.chunk.shape;
+        this.positions = this.shape.positions;
+        this.moveToPosition(0);
+    }
+
+    private moveToPosition(positionIndex: number): void {
+        this.nodePositionInfo = this.positions[positionIndex];
+        this.positionIndex = positionIndex;
+    }
+
+    nextField(): boolean {
+        this.indexOfField++;
+        const fields = this.nodeInfo(CursorLocationType.Fields).shape.fieldsArray;
+        if (this.indexOfField < fields.length) {
+            this.fieldKey = fields[this.indexOfField][0];
+            return true;
+        }
+        this.exitField();
+        return false;
+    }
+
+    exitField(): void {
+        assert(this.mode === CursorLocationType.Fields, "exitField when in wrong mode");
+        this.fieldKey = undefined;
+        this.mode = CursorLocationType.Nodes;
+    }
+
+    getFieldKey(): FieldKey {
+        return this.fieldKey ?? fail("not in a field");
+    }
+
+    getFieldLength(): number {
+        const info = this.nodeInfo(CursorLocationType.Fields);
+        const fieldInfo = info.shape.fieldsArray[this.indexOfField];
+        if (fieldInfo === undefined) {
+            return 0;
+        }
+        return fieldInfo[2];
+    }
+
+    firstNode(): boolean {
+        const info = this.nodeInfo(CursorLocationType.Fields);
+        const fields = info.shape.fieldsArray;
+        if (this.indexOfField >= fields.length) {
+            return false; // Handle empty field (indexed by key into empty field)
+        }
+        this.enterNode(0); // TODO: perf: this redoes some lookups
+        return true;
+    }
+
+    enterNode(childIndex: number): void {
+        const info = this.nodeInfo(CursorLocationType.Fields);
+        const f = info.shape.fieldsOffsetArray[this.indexOfField];
+        assert(childIndex >= 0, "index past end of field");
+        assert(childIndex < f.topLevelLength, "index past end of field");
+        this.mode = CursorLocationType.Nodes;
+        this.moveToPosition(this.positionIndex + f.offset + childIndex * f.shape.positions.length);
+        assert(this.fieldIndex === childIndex, "should be at selected child");
+    }
+
+    getFieldPath(): FieldUpPath {
+        return {
+            field: this.getFieldKey(),
+            parent: this.nodeInfo(CursorLocationType.Fields),
+        };
+    }
+
+    getPath(): UpPath | undefined {
+        return this.nodeInfo();
+    }
+
+    get fieldIndex(): number {
+        return this.nodeInfo().parentIndex;
+    }
+
+    public get chunkStart(): number {
+        // TODO: properly report chunks and expose fast path
+        return this.fieldIndex;
+    }
+
+    public get chunkLength(): number {
+        // TODO: properly report chunks and expose fast path
+        return 1;
+    }
+
+    seekNodes(offset: number): boolean {
+        const info = this.nodeInfo(CursorLocationType.Nodes);
```

Comment:
It doesn't seem like nodeInfo is ever used with in Fields mode and this is the only time it's passed in explicitly. Can we just remove this parameter?

Reply:
5 of the 13 usages in are in fields mode. Since both uses are about as common, I've removed the default and added better docs.